connector/src/main/java/org/jboss/as/connector/subsystems/datasources/Constants.java

```diff
@@ -227,18 +227,15 @@
     static final SensitiveTargetAccessConstraintDefinition DS_SECURITY_DEF = new SensitiveTargetAccessConstraintDefinition(DS_SECURITY);
 
     static final SimpleAttributeDefinition DEPLOYMENT_NAME = SimpleAttributeDefinitionBuilder.create("deployment-name", ModelType.STRING)
-            .setAllowExpression(true)
             .setRequired(false)
             .build();
 
     static final SimpleAttributeDefinition MODULE_SLOT = SimpleAttributeDefinitionBuilder.create("module-slot", ModelType.STRING)
-            .setAllowExpression(true)
```

Comment:
This and the change to DRIVER_MODULE_NAME are breaking changes. AIUI what's driving this change is parsing the xml, distinguishing between the ':'  separating the name and slot vs a ':' that may be inside an expression. If so that seems like something that can be solved without a breaking change.

Reply:
I have added the new parsing and updated the test.